tests/test_cli_compile.py

```diff
@@ -2100,6 +2100,35 @@ def test_multiple_extras(fake_dists, runner, make_module, fname, content, extra_
     assert "extra ==" not in out.stderr
 
 
+@pytest.mark.network
+@pytest.mark.parametrize(("fname", "content"), METADATA_TEST_CASES)
+@pytest.mark.xfail(is_pypy, reason="https://github.com/jazzband/pip-tools/issues/1375")
+def test_wildcard_extras(fake_dists, runner, make_module, fname, content):
+    """
+    Test passing wildcard `--extra` includes all applicable extras.
+    """
+    meta_path = make_module(fname=fname, content=content)
+    out = runner.invoke(
+        cli,
+        [
+            "-n",
+            "--extra",
+            "*",
+            "--find-links",
+            fake_dists,
+            meta_path,
+        ],
+    )
+    assert out.exit_code == 0, out.stderr
+    assert "small-fake-a==0.1" in out.stderr
+    assert "small-fake-b==0.2" in out.stderr
+    assert "small-fake-c==0.3" in out.stderr
+    assert "small-fake-d==0.4" in out.stderr
+    assert "small-fake-e==0.5" in out.stderr
+    assert "small-fake-f==0.6" in out.stderr
+    assert "extra ==" not in out.stderr
```

Comment:
```suggestion
    assert out.stderr == expected
```

Could you please check the whole output?

Reply:
Certainly. Do you think it is better to save the output to a file and read it or as a string literal in the test?
A third idea is to implement a parser, if there is not one already, and do the comparison in the parsed domain. This way the formatting, and any future changes to it, does not affect the test.

Should we also update similar tests such as `test_multiple_extras`?

Reply:
Sorry for leaving this unfinished. There's another test that uses a string literal so that is what I did as well. Anything else that can be improved?

Reply:
It's all good now! Thanks 👍🏻